### PR TITLE
Fix: prefer-destructuring invalid autofix with computed property access

### PR DESCRIPTION
**Pull request by mdjermanovic**
#### Prerequisites checklist

- [X] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/master/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[ ] Documentation update
[X] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-change-proposal.md))
[ ] Add autofixing to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

**Tell us about your environment**

* **ESLint Version:** v7.9.0
* **Node Version:** v12.18.4
* **npm Version:** v6.14.6

**What parser (default, `@babel/eslint-parser`, `@typescript-eslint/parser`, etc.) are you using?**

default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
  parserOptions: {
    ecmaVersion: 2015
  }
};
```

</details>

**What did you do? Please include the actual source code causing the issue.**

[Online Demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IHByZWZlci1kZXN0cnVjdHVyaW5nOiBbXCJlcnJvclwiLFxuICB7IFwib2JqZWN0XCI6IHRydWUgfSxcbiAgeyBcImVuZm9yY2VGb3JSZW5hbWVkUHJvcGVydGllc1wiOiB0cnVlIH1cbl0gKi9cblxudmFyIGEgPSBiW2FdOyIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6Niwic291cmNlVHlwZSI6InNjcmlwdCIsImVjbWFGZWF0dXJlcyI6e319LCJydWxlcyI6e30sImVudiI6e319fQ==)

```js
/* eslint prefer-destructuring: ["error",
  { "object": true },
  { "enforceForRenamedProperties": true }
] */

var a = b[a];
```

**What did you expect to happen?**

1 error, but not with an invalid autofix.

**What actually happened? Please include the actual, raw output from ESLint.**

1 error and autofix to:

```js
/* eslint prefer-destructuring: ["error",
  { "object": true },
  { "enforceForRenamedProperties": true }
] */

var {a} = b;
```

This changes the semantics. A correct autofix would be `var { [a]: a } = b;`

#### What changes did you make? (Give an overview)

Fixed the `prefer-destructuring` rule to not auto-fix in this case, since the rule generally doesn't auto-fix renamed properties. This was just mistakenly treated as if it was `var a = b.a;`.

### CODE REVIEW

**Review by nzakas — CHANGES_REQUESTED** [review]
Can you add something about this to the docs too? It seems like this new behavior could be misinterpreted as a bug without an explanation somewhere.

**Comment by mdjermanovic** [comment]
Added a paragraph explaining what code is fixable by this rule.

This PR doesn't change behavior much, the rule already fixes only simple cases:

```js
/* eslint prefer-destructuring: ["error",
  { "object": true },
  { "enforceForRenamedProperties": true }
] */ 

var foo = object.foo; // auto-fixable

var foo = object.bar; // not auto-fixable

var foo = object[bar]; // not auto-fixable

var foo = object[foo]; // was mistakenly auto-fixable
```

[Demo link](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IHByZWZlci1kZXN0cnVjdHVyaW5nOiBbXCJlcnJvclwiLFxuICB7IFwib2JqZWN0XCI6IHRydWUgfSxcbiAgeyBcImVuZm9yY2VGb3JSZW5hbWVkUHJvcGVydGllc1wiOiB0cnVlIH1cbl0gKi8gXG5cbnZhciBmb28gPSBvYmplY3QuZm9vOyAvLyBhdXRvLWZpeGFibGVcblxudmFyIGZvbyA9IG9iamVjdC5iYXI7IC8vIG5vdCBhdXRvLWZpeGFibGVcblxudmFyIGZvbyA9IG9iamVjdFtiYXJdOyAvLyBub3QgYXV0by1maXhhYmxlXG5cbnZhciBmb28gPSBvYmplY3RbZm9vXTsgLy8gd2FzIG1pc3Rha2VubHkgYXV0by1maXhhYmxlXG4iLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOjYsInNvdXJjZVR5cGUiOiJzY3JpcHQiLCJlY21hRmVhdHVyZXMiOnt9fSwicnVsZXMiOnt9LCJlbnYiOnt9fX0=)

**Review by nzakas — APPROVED** [review]
That works. Thanks.

File: docs/rules/prefer-destructuring.md [hunk 1]
```diff
@@ -21,6 +21,8 @@ The rule has a second object with a single key, `enforceForRenamedProperties`, w
 - Accessing an object property whose key is an integer will fall under the category `array` destructuring.
 - Accessing an array element through a computed index will fall under the category `object` destructuring.
 
+The `--fix` option on the command line fixes only problems reported in variable declarations, and among them only those that fall under the category `object` destructuring. Furthermore, the name of the declared variable has to be the same as the name used for non-computed member access in the initializer. For example, `var foo = object.foo` can be automatically fixed by this rule. Problems that involve computed member access (e.g., `var foo = object[foo]`) or renamed properties (e.g., `var foo = object.bar`) are not automatically fixed.
```
Comment (btmills) [inline docs/rules/prefer-destructuring.md:24]: Adding this also covers the `var foo = object['foo'];` case, which we don't fix either and for which we already have a test asserting the same.

**Review by btmills — APPROVED** [review]

**Pull request merged**